entities/src/main/java/org/odk/collect/entities/storage/Entity.kt

```diff
@@ -35,9 +35,15 @@ sealed interface Entity {
         override val branchId: String = UUID.randomUUID().toString()
     ) : Entity
 
-    enum class State {
-        OFFLINE,
-        ONLINE
+    enum class State(val id: Int) {
+        OFFLINE(0),
```

Comment:
Wouldn't it be easier to just use the string representation instead of numbers? If you call `toString`  on `State.OFFLINE` you should get `OFFLINE` (the same with `State.ONLINE`). You can store such a string in the database.

Reply:
I think that would add quite a lot more data though as you'd end up with "OFFLINE" or "ONLINE" in every entity row (which we're hoping we can get into the millions).

Reply:
Oh interesting. Could you add a comment explaining that? I'm pretty sure I would be tempted to refactor that code next time I work with it.

Reply:
Sure! I'll also move the conversion to `DatabaseEntitiesRepository` as `Entity` shouldn't really need to know anything about "serializing" the enum.